src/main/java/com/hpe/adm/octane/ideplugins/services/util/UrlParser.java

```diff
@@ -93,6 +93,12 @@ private static ConnectionSettings resolveConnectionSettingsFromUrl(String url) t
                     baseUrl += ":" + siteUrl.getPort();
                 }
 
+                String siteUrlPath = siteUrl.getPath();
+                if (siteUrlPath.endsWith("/ui/"))
+                    baseUrl += siteUrlPath.substring(0, siteUrlPath.length() - 4);
```

Comment:
this is a good place to document why you are using a substring. Or maybe use the /api  constant throughout the code and use here it's length